forest/jasmine/mobmat2traj.py

```diff
@@ -1,570 +1,1327 @@
+            if weight is None:
+                sys.exit("Invalid method for calculate_k1.")
+
+            normalize_w = (weight + 1e-5) / float(sum(weight + 1e-5))
+            pause_index = np.random.choice(pause_table.shape[0], p=normalize_w)
+            delta_t = (
+                pause_table[pause_index, 6] - pause_table[pause_index, 3]
+            ) * multiplier(end_t - start_t)
+
+            if start_t + delta_t < end_t:
+                current_t = start_t + delta_t
+                imp_table = update_table(
+                    imp_table,
+                    [
+                        2, start_x, start_y,
+                        start_t, start_x,
+                        start_y, current_t
+                    ]
+                )
+                start_t, start_s = current_t, 2
+                counter += 1
+            else:
+                imp_table = update_table(
+                    imp_table,
+                    [
+                        1, start_x, start_y,
+                        start_t, end_x,
+                        end_y, end_t
+                    ]
+                )
+                start_t = end_t
+    return imp_table, (start_s, start_t, start_x, start_y), counter
+
+
+def backward_impute(
+    end_t: float, end_x: float, end_y: float,
+    start_t: float, start_x: float, start_y: float,
+    bv_set: np.ndarray, switch: int, num: int, pars: list,
+    flight_table: np.ndarray, linearity: float,
+    mis_row: np.ndarray, pause_table: np.ndarray,
+    imp_table: np.ndarray, end_s: int, method: str,
+    counter: int
+) -> Tuple[np.ndarray, Tuple[int, float, float, float], int]:
+    """
+    This function imputes a missing interval
+        from the end point to the start point.
+
+    Args:
+        end_t: float, end time
+        end_x, end_y: float, end position
+        start_t: float, start time
+        start_x, start_y: float, start position
+        bv_set: np.ndarray, output from BV_select()
+        switch: int, the number of binary variables to be generated
+        num: int, checks the top k similarities
+        pars: list, the parameters that are required
+         for the calculate_k1 function
+        flight_table: np.ndarray, output from create_tables()
+        linearity: float, controls the smoothness of a trajectory
+        mis_row: np.ndarray, a row of missing interval table
+        pause_table: np.ndarray, output from create_tables()
+        imp_table: np.ndarray, output from create_tables()
+        end_s: int, status of the end point
+        method: str, the method to be used for calculation,
+         should be either 'TL', 'GL', or 'GLC'
+        counter: int, number of imputed trajectories
+
+    Returns:
+        imp_table: updated imp_table
+        tuple of 4 elements:
+            end_s: int, status of the end point
+            end_t: float, end time
+            end_x, end_y: float, end position
+        counter: updated counter
+    """
+
+    I1 = indicate_flight(
```

Comment:
Variable in function should be lowercase. Also, a more descriptive name would make the code more readable.